Build failure: percona-server

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

`nix build github:nixos/nixpkgs/nixpkgs-unstable#percona-server`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/287017684

### Relevant log output

```console
[  3%] Building CXX object libs/mysql/serialization/CMakeFiles/mysql_serialization.dir/archive_binary.cpp.o
In file included from /tmp/nix-build-percona-server-8.4.3-3.drv-0/percona-server-8.4.3-3/libs/mysql/serialization/archive_binary.cpp:24:
In file included from /tmp/nix-build-percona-server-8.4.3-3.drv-0/percona-server-8.4.3-3/libs/mysql/serialization/archive_binary.h:29:
/tmp/nix-build-percona-server-8.4.3-3.drv-0/percona-server-8.4.3-3/libs/mysql/serialization/archive.h:84:43: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
   84 |     return Archive_derived_type::template get_size(std::forward<Type>(arg));
      |                                           ^
/tmp/nix-build-percona-server-8.4.3-3.drv-0/percona-server-8.4.3-3/libs/mysql/serialization/archive.h:91:43: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
   91 |     return Archive_derived_type::template get_size_written();
      |                                           ^
2 errors generated.
make[2]: *** [libs/mysql/serialization/CMakeFiles/mysql_serialization.dir/build.make:79: libs/mysql/serialization/CMakeFiles/mysql_serialization.dir/archive_binary.cpp.o] Error 1
```

### Additional context

Should be fixed in the next release: https://perconadev.atlassian.net/browse/PS-9542

### System metadata

darkkeks:~$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.6.0, macOS 14.7.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.10`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/djws0nzaa2g7xz9iv90ja8yxqddm37yy-source`

### Notify maintainers

@osnyx @leona-ya 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/servers/sql/percona-server/8_0.nix
+++ b/pkgs/servers/sql/percona-server/8_0.nix
@@ -142,7 +142,7 @@ stdenv.mkDerivation (finalAttrs: {
   postInstall = ''
     moveToOutput "lib/*.a" $static
     so=${stdenv.hostPlatform.extensions.sharedLibrary}
-    ln -s libmysqlclient$so $out/lib/libmysqlclient_r$so
+    ln -s libperconaserverclient$so $out/lib/libmysqlclient_r$so
 
     wrapProgram $out/bin/mysqld_safe --prefix PATH : ${
       lib.makeBinPath [

--- a/pkgs/servers/sql/percona-server/8_4.nix
+++ b/pkgs/servers/sql/percona-server/8_4.nix
@@ -162,7 +162,7 @@ stdenv.mkDerivation (finalAttrs: {
     ''
       moveToOutput "lib/*.a" $static
       so=${stdenv.hostPlatform.extensions.sharedLibrary}
-      ln -s libmysqlclient$so $out/lib/libmysqlclient_r$so
+      ln -s libperconaserverclient$so $out/lib/libmysqlclient_r$so
 
       wrapProgram $out/bin/mysqld_safe --prefix PATH : ${
         lib.makeBinPath [

